projectile.el

```diff
@@ -5466,20 +5466,28 @@ An open project is a project with any open buffers."
        (list (abbreviate-file-name project)))
     projects))
 
+(defun projectile--move-current-project-to-front (projects)
+  "Move current project (if any) to the start of list in the list of PROJECTS."
+  (if-let ((project (projectile-project-root)))
+      (append
+       (list (abbreviate-file-name project))
+       (projectile--remove-current-project projects))
+    projects))
+
 (defun projectile-relevant-known-projects ()
   "Return a list of known projects."
   (pcase projectile-current-project-on-switch
     ('remove (projectile--remove-current-project projectile-known-projects))
     ('move-to-end (projectile--move-current-project-to-end projectile-known-projects))
-    ('keep projectile-known-projects)))
+    ('keep (projectile--move-current-project-to-front projectile-known-projects))))
```

Comment:
Maybe we should have `move-to-front` here, as `keep` is kind of supposed to keep things as they are.

Reply:
As a first step I implemented this suggestion but overhauling the find-file-hooks logic sounds like a pretty nice solution. I'll look into that but m free time has recently been swept away so no promises on timeline for that